cns/restserver/ipam.go

```diff
@@ -713,8 +713,13 @@ func (service *HTTPRestService) releaseIPConfigs(podInfo cns.PodInfo) error {
 	service.Lock()
 	defer service.Unlock()
 	ipsToBeReleased := make([]cns.IPConfigurationStatus, 0)
-
-	for i, ipID := range service.PodIPIDByPodInterfaceKey[podInfo.Key()] {
+	key := podInfo.Key()
+	if _, isMapContainsKey := service.PodIPIDByPodInterfaceKey[podInfo.Key()]; !isMapContainsKey {
+		// the special case for azure CNI with managed endpoint state since the podInfo.Key() is in containeID-eth0 format and we need to use full ContainerID as well
```

Comment:
why can't we change podInfo.Key() format to containerid in same way as delegated azure ipam

Reply:
I was wondering this as well.. in Cilium azure-ipam we are setting `args.ContainerID` for both `PodInterfaceID` and `InfraContainerID`, is there a reason why we can't do it in stateless CNI?

Reply:
@tamilmani1989 @nddq Addressed in the new commit